velox/functions/lib/Re2Functions.cpp

```diff
@@ -817,6 +1070,87 @@ std::shared_ptr<VectorFunction> makeRe2Extract(
   }
 }
 
+std::vector<std::shared_ptr<exec::FunctionSignature>> re2ReplaceSignatures() {
+  return {
+
+      // regex_replace (str, pattern, replace)
+      exec::FunctionSignatureBuilder()
+          .returnType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .build(),
+      // regex_replace (str, pattern, replace, position)
+      exec::FunctionSignatureBuilder()
+          .returnType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("integer")
+          .build(),
+      // regex_replace (str, pattern, replace, position)
+      exec::FunctionSignatureBuilder()
+          .returnType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("varchar")
+          .argumentType("bigint")
+          .build(),
+  };
+}
+
+std::shared_ptr<VectorFunction> makeRe2Replace(
+    const std::string& name,
+    const std::vector<VectorFunctionArg>& inputArgs,
+    const core::QueryConfig& /*config*/
+) {
+  auto numArgs = inputArgs.size();
+
+  VELOX_USER_CHECK(
+      numArgs == 3 || numArgs == 4,
+      "{} requires between 2 and 4 arguments, but got {}",
+      name,
+      numArgs);
+
+  VELOX_USER_CHECK(
```

Comment:
loop over the three arguments and check?

Reply:
Doable but also not sure if necessary. Loop would need to handle the grammar of first, second, third, fourth, etc. Additionally, no other functions seem to handle it this way (with a loop). Thoughts?